### PR TITLE
I7369-setting up telemetry reducer

### LINKED ISSUE #7369: Set up telemetry reducer (for TAAR in disco)
This is to support work for #7218 

(REF comment: https://github.com/mozilla/addons-frontend/pull/7357#discussion_r247159067)

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #7369

### CODE REVIEW

File: src/disco/reducers/telemetry.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+type SetHashedClientIdAction = {|
+  type: typeof SET_HASHED_CLIENT_ID,
+  payload: { hashedClientId: string },
+|};
+
+export const setHashedClientId = (
+  hashedClientId: string,
+): SetHashedClientIdAction => {
+  if (!hashedClientId) {
+    invariant(hashedClientId, 'hashedClientId is required');
```
Comment (willdurand) [inline src/disco/reducers/telemetry.js]: Why do we need the if here?

File: src/disco/reducers/telemetry.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* @flow */
+import invariant from 'invariant';
+
+export const SET_HASHED_CLIENT_ID: 'SET_HASHED_CLIENT_ID' =
+  'SET_HASHED_CLIENT_ID';
+
+export type TelemetryState = {|
+  hashedClientId?: string,
```
Comment (willdurand) [inline src/disco/reducers/telemetry.js]: Maybe we could make it non-optional and nullable?

File: src/disco/reducers/telemetry.js (hunk 1 above)
Comment (rebmullin) [reply src/disco/reducers/telemetry.js]: ok I can remove this. I just thought it was an extra check but maybe it's not necessary..

File: src/disco/reducers/telemetry.js (hunk 2 above)
Comment (rebmullin) [reply src/disco/reducers/telemetry.js]: I personally prefer undefined if it's not set but I can default the state to null.

File: src/disco/reducers/telemetry.js (hunk 1 above)
Comment (willdurand) [reply src/disco/reducers/telemetry.js]: invariant should be enough

Comment (rebmullin) [reply src/disco/reducers/telemetry.js]: ahhh I see:)

**Pull request merged**